shared/protocol/src/main/java/io/pravega/shared/protocol/netty/CommandEncoder.java

```diff
@@ -88,66 +86,57 @@ protected void encode(ChannelHandlerContext ctx, Object msg, ByteBuf out) throws
         log.trace("Encoding message to send over the wire {}", msg);
         if (msg instanceof Append) {
             Append append = (Append) msg;
+            if (!append.segment.equals(segmentBeingAppendedTo) || !append.getWriterId().equals(writerIdPerformingAppends)) {
```

Comment:
The description of this PR says:

```
... the class uses the getAppendBatchSizeTracker method supplied by FlowHandler to identify the appendBatchSizeTracker for append requestid(flowid).
```

but I don't see it being used here. There is a discrepancy between what this is doing and what is described in the PR.

Reply:
The encode method of the CommandEncoder does the following:
           if (appendTracker != null) {
                blockSizeSupplier = appendTracker.apply(append.getFlowId());
                blockSizeSupplier.recordAppend(append.getEventNumber(), msgSize);
            }
getAppendBatchSizeTracker  is supplied and appendTracker.apply inovkes the method getAppendBatchSizeTracker  to retrieve the append batch size tracker for the given flowid.